Firebase

lattln opened the pull request:
+ firebase setup
+ log in session
+ private / protected routes
+ Logos & backgrounds
+ multiple routes & Pages

lattln committed:
firebase

+ firebase setup
+ log in session
+ private / protected routes
+ Logos & backgrounds
+ multiple routes & Pages

lattln committed:
fix

+ rename
+ sticky navbar
+ route fixes

Nguyen-Noah commented on macro-and-me/src/pages/loginPage/Login.jsx:
```diff
@@ -0,0 +1,128 @@
+    const [signUp, setSignUp] = useState(false);
+
+    const handleAuth = () => {
+        auth.setPersistence(firebase.auth.Auth.Persistence.SESSION)  
```
Can we change this to `setPersistence(auth, browserSessionPersistence)`? It acts the same as `auth.setPersistence(firebase.auth.Auth.Persistence.SESSION)`, but it's more modular on firebase SDK >9.x.x.

lattln replied:
yes sir 07

lattln committed:
browser session

+ fixed browser session
+ modular persistence
+ sign-in with popup for handling google sign-in
+ Firebase SDK version >9.x.x

lattln committed:
Merge branch 'firebase' of https://github.com/Nguyen-Noah/macro-and-me into firebase

myeken reviewed (commented):
Everything looks good to me.
